components/datetime/src/neo.rs

```diff
@@ -229,6 +265,78 @@ pub struct TypedNeoFormatter<C: CldrCalendar, R: DateTimeNamesMarker> {
     _calendar: PhantomData<C>,
 }
 
+impl<C: CldrCalendar, FSet: DateTimeMarkers + HasConstComponents> TypedNeoFormatter<C, FSet>
+where
+    FSet::D: TypedDateDataMarkers<C>,
+    FSet::T: TimeMarkers,
+    FSet::Z: ZoneMarkers,
+    FSet: NeoGetField<FSet::LengthOption>,
+    FSet: NeoGetField<FSet::AlignmentOption>,
+    FSet: NeoGetField<FSet::EraDisplayOption>,
+    FSet: NeoGetField<FSet::FractionalSecondDigitsOption>,
+{
+    /// Creates a new [`TypedNeoFormatter`] from compiled data with
+    /// datetime components specified at build time.
+    ///
+    /// Use this constructor for optimal data size and memory use
+    /// if you know the required datetime components at build time.
+    /// If you do not know the datetime components until runtime,
+    /// use a `with_components` constructor.
+    ///
+    /// # Examples
+    ///
+    /// Basic usage:
+    ///
+    /// ```
+    /// use icu::calendar::Date;
+    /// use icu::calendar::Gregorian;
+    /// use icu::datetime::neo::TypedNeoFormatter;
+    /// use icu::datetime::neo_marker::NeoYearMonthDayMarker;
+    /// use icu::datetime::neo_skeleton::NeoSkeletonLength;
+    /// use icu::locale::locale;
+    /// use writeable::assert_try_writeable_eq;
+    ///
+    /// let formatter =
+    ///     TypedNeoFormatter::try_new2(
+    ///         &locale!("es-MX").into(),
+    ///         NeoYearMonthDayMarker::with_length(NeoSkeletonLength::Long),
+    ///     )
+    ///     .unwrap();
+    ///
+    /// assert_try_writeable_eq!(
+    ///     formatter.format(&Date::try_new_gregorian_date(2023, 12, 20).unwrap()),
+    ///     "20 de diciembre de 2023"
+    /// );
+    /// ```
+    #[cfg(feature = "compiled_data")]
+    pub fn try_new2(locale: &DataLocale, options: FSet) -> Result<Self, LoadError>
```

Comment:
issue (nb): don't like this name, but we can revisit later

Reply:
I intend to replace the old `try_new` with the new signature

Reply:
sgtm